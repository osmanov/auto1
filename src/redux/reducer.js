import { combineReducers } from 'redux'
import carsReducer, { moduleName as carsModule } from '../ducks/cars'
import carReducer, { moduleName as carModule } from '../ducks/car'
import propertiesReducer, {
  moduleName as propertiesModule
} from '../ducks/properties'

export default combineReducers({
  [carsModule]: carsReducer,
  [carModule]: carReducer,
  [propertiesModule]: propertiesReducer
})
