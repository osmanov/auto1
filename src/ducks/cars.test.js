import { put } from 'redux-saga/effects'
import reducer, {
  initialState,
  fetchCarsSaga,
  FETCH_CARS_SUCCESS,
  FETCH_CARS_ERROR
} from './cars'

const mock = {
  cars: [
    {
      color: 'yellow',
      fuelType: 'Petrol',
      manufacturerName: 'BMW',
      mileage: {},
      modelName: 'X5',
      pictureUrl: 'http://localhost:3001/car.svg',
      stockNumber: 10025
    }
  ]
}
const state = Object.freeze(initialState)
/**
 * Reducer Tests
 */
it('reducer should get cars', () => {
  const newState = reducer(state, { type: FETCH_CARS_SUCCESS, payload: mock })
  expect(newState).toMatchSnapshot()
})
/**
 * Saga Tests
 */
it('saga should get cars', () => {
  const saga = fetchCarsSaga()
  expect(saga.next().value).toMatchSnapshot()
  expect(saga.next({ payload: { page: 1 } }).value).toMatchSnapshot()
  expect(saga.next(mock).value).toEqual(
    put({
      type: FETCH_CARS_SUCCESS,
      payload: mock
    })
  )

  const error = new Error()

  expect(saga.throw(error).value).toEqual(
    put({
      type: FETCH_CARS_ERROR,
      error
    })
  )
})
