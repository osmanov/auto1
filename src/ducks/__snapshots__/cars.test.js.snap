// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer should get cars 1`] = `
Object {
  "data": Array [
    Object {
      "color": "yellow",
      "fuelType": "Petrol",
      "manufacturerName": "BMW",
      "mileage": Object {},
      "modelName": "X5",
      "pictureUrl": "http://localhost:3001/car.svg",
      "stockNumber": 10025,
    },
  ],
  "error": null,
  "filter": null,
  "loading": false,
  "sort": null,
  "totalPageCount": 0,
}
`;

exports[`saga should get cars 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "pattern": "auto1/cars/FETCH_CARS_REQUEST",
  },
  "type": "TAKE",
}
`;

exports[`saga should get cars 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/cars?&page=1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
